WolvenKit.App/Factories/ChunkViewmodelFactory.cs

```diff
@@ -0,0 +1,106 @@
+﻿using WolvenKit.App.Controllers;
+using WolvenKit.App.Models.Nodify;
+using WolvenKit.App.ViewModels.Shell;
+using WolvenKit.Common.Services;
+using WolvenKit.Common;
+using WolvenKit.Core.Interfaces;
+using WolvenKit.RED4.CR2W;
+using WolvenKit.RED4.Types;
+using WolvenKit.App.ViewModels.Documents;
+using WolvenKit.App.Services;
+
+namespace WolvenKit.App.Factories;
+
+public interface IFactory<T> where T : class
```

Comment:
What I meant previously was that this should be something like https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.ilogger?view=dotnet-plat-ext-7.0. With a default method for creation, then where service is registered you can have a generic concrete class that creates an instance of `T`.